doc/examples/boilerplate.py

```diff
@@ -0,0 +1,135 @@
+"""
+A docstring
+
+This can double as a helpful message which explains how the program is run.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import sys
+from typing import List  # noqa: F401  Flake8 doesn't recognise Python 2 typing.
+
+import libtbx.phil
+from dials.array_family import flex
+import dials.util
+import dials.util.log
+from dials.util.options import OptionParser, flatten_reflections, flatten_experiments
+from dxtbx.model import ExperimentList
+
+
+# Define a logger
+logger = logging.getLogger("dials.logger_name")
+
+# Define the master PHIL scope for this program
+phil_scope = libtbx.phil.parse(
+    """
+    output {
+        reflections = stronger.refl
+            .type = path
+            .help = "Help strings are helpful, wherever possible."
+        log = dials.command_name.log
+            .type = path
+    }
+    bool_parameter = False
+      .type = bool
+    integer_parameter = 0
+      .type = int
+    """
+)
+
+
+def do_stuff(experiments, reflections, params):
+    # type: (ExperimentList, flex.reflection_table, libtbx.phil.scope_extract) -> None
+    """
+    Write the behaviour of the program as functions and classes outside run().
+
+    Don't include file output here, remember that this function may be re-used
+    elsewhere by someone who doesn't need the output written immediately to file.
+
+    It can be especially helpful to document any expected exceptions that might be
+    raised, in order to keep track of what needs to be handled in any code that
+    re-uses this function.
+
+    Args:
+        experiments:  An experiment list.
+        reflections:  A reflection table.
+        params:       Some parameters, in the form of a scope_extract object,
+                      which is the usable form of a parsed PHIL scope.
+
+    Raises:
+        RuntimeError:  if you pass an empty experiment list.
+    """
+    logger.info("Hello world!")
+
+    # Here's an example of an error that might be raised, as documented above.
+    if not experiments:
+        raise RuntimeError
```

Comment:
should be in `run()` method?

Reply:
Yes, I agree, I was looking to provide a simple example of a raise that needs documenting in the docstring, but this is probably a bad example.